iamhectorsosa/supabase-modules
# feat: Anonymous user access

## Summary

This PR implements / improves the following **features**:

- #32 - guest login

This changes routes file hierarchy a bit by introducing [a grouped route](https://nextjs.org/docs/app/building-your-application/routing/route-groups) (`(authenticated)`) so the application layout can be shared between the routes rather than repeatable specify it in each `layout.tsx`.

chores done:

Removes some cache files that got into the versioned code.

## Testing

1. Go to login page and click on **Continue as guest**
2. You should be presented a Guest section after login where finalizing registration is possible
3. There should be only bookmarks accessible at this point from within the layout (aside navigation + dropdown)
4. Type an email address and hit **Finish registration**, you should be presented with the instructions to confirm in the mailbox
5. After confirming the email there should be another view (but still the Guest section only) with the form to set a password, another sections of app are still inaccessible
6. Fill in a password and hit **Set password**
7. You should be now successfully converted to a permanent user and Settings section of the app should be accessible 


## Related issues

closes #32
